### PR TITLE
Loki: Remove hardcoded values for parsed query parts

### LINKED ISSUE #53116: Loki: Remove hardcoded values for parsed query parts
**Level: easy**

 In public/app/plugins/datasource/loki/query_utils.ts and in lezer-logql we are using many string comparisons. Example: 
	if (type.name === 'LabelFormatExpr') {
   		queryWithLabelFormat = true;
}

	Change those to something like 
	if (type.id === LabelFormatExpr) {

This can be be also done for Prometheus and lezer-promql.

### LINKED ISSUE #55024: Loki:  Refactor use of lezer nodes to reference ID instead of name in parsing.ts
Example of how this was done for Prometheus: https://github.com/grafana/grafana/pull/53177/files. 

In `public/app/plugins/datasource/loki/querybuilder/parsing.ts` instead of using hard-coded names of nodes (e.g. `LabelParser` ), we can use imported types from parser.

Example: 
```
switch (node.name) {
    case 'LabelParser': {
```

Can be changed to something like (where `MetricIdentifier` is imported from the parser)

```
switch (node.type.id) {
    case LabelParser: {
```

The only `id variable`  I couldn't find is `Error` - it has id of `0`, but I couldn't find its exported name. Not sure if we can replace this hard-coded value, but other should be straight forward.

### PR DESCRIPTION
**Pull request by gitstart**
**What this PR does / why we need it**:
Removes hardcoded values for parsed query parts

**Which issue(s) this PR fixes**:

Fixes #53116
Fixes: #55024

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by gitstart** [comment]
Hi @matyax, please review the PR

**Review by ivanahuckova — COMMENTED** [review]
Thank you for your contribution! I've reviewed the code and: 

-  In`public/app/plugins/datasource/loki/modifyQuery.ts` there are still some hard coded strings. Specifically in `getLogQueryPositions` when calling `node.getChild()`. The harcoded strings should be changed to imported ids (`'Selector'` -> `Selector`)

- in `public/app/plugins/datasource/loki/query_utils.ts` the same thing in `isQueryPipelineErrorFiltering`

**Comment by gitstart** [comment]
@ivanahuckova Thank you for the review, will get back to you with the requested changes

**Comment by gitstart** [comment]
@ivanahuckova Requested changes has been made, please take a look

**Comment by gitstart** [comment]
@ivanahuckova! removed some more of the hardcoded values, please take a look 🙏

File: public/app/plugins/datasource/prometheus/querybuilder/shared/parsingUtils.ts [hunk 1]
```diff
@@ -2,8 +2,8 @@ import { SyntaxNode, TreeCursor } from '@lezer/common';
 
 import { QueryBuilderOperation } from './types';
 
-// This is used for error type for some reason
-export const ErrorName = '⚠';
+// Although 0 isn't explicitly provided in the lezer-promql library as the error node ID, it does appear to be the ID of error nodes within lezer.
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/prometheus/querybuilder/shared/parsingUtils.ts:5]: I'd mentioned here both `lezer-promql` and `@grafana/lezer-logql`: 

`... in the lezer-promql and @grafana/lezer-logql libraries as the error node ID...`

**Review by ivanahuckova — COMMENTED** [review]
I've left feedback with minor suggested changes. Otherwise LGTM.

**Comment by gitstart** [comment]
Thank you for the feedback @ivanahuckova. The changes has been made, please take a look

**Comment by ivanahuckova** [comment]
This was affected by https://github.com/grafana/grafana/pull/55180. You need to resolve merge conflicts. Otherwise LGTM!

**Review by ivanahuckova — COMMENTED** [review]
This was affected by https://github.com/grafana/grafana/pull/55180. You need to resolve merge conflicts. Otherwise LGTM!

**Comment by gitstart** [comment]
@ivanahuckova merge conflict has been resolved now

**Review by ivanahuckova — APPROVED** [review]
LGMT!

**Pull request merged**